src/smc-webapp/project_store.coffee

```diff
@@ -1200,15 +1200,17 @@ class ProjectActions extends Actions
                 if target[target.length-1] == '/' or full_path == ''
                     @open_directory(parent_path)
                 else
-                    @fetch_directory_listing(parent_path)
+                    @fetch_directory_listing(parent_path, true, true)
                     @get_store().wait
-                        until   : (s) => s.directory_listings.get(parent_path)
+                        until   : (s) =>
+                            listing = s.directory_listings.get(parent_path)
+                            return listing?.find (val) => val.get('name') == last
```

Comment:
Just a comment -- no need to do anything now:   It seems like this line is doing a linear search through a potentially large data structure, which is clearly not the optimal way to structure our data -- instead shouldn't there be a map, so it's a O(log(N)) search.